src/plugins/ClausePlugin.js

```diff
@@ -207,6 +207,15 @@ function ClausePlugin() {
     && (node.data.get('clauseid') === clauseId));
   }
 
+  /**
+   * Is the clause UI-supported
+   */
+  function isClauseSupported(editor, clauseId) {
+    const hasConditionals = (node) => { return node.type === 'conditional'; }
+    const unsupportedNodes = editor.value.document.nodes.find(node => hasConditionals(node));
```

Comment:
Since the clause node will not be on the document yet (we want to prevent adding it if it's unsupported), we won't be able to find the node this way. I think we want to pass the whole clause node to this method instead of `clauseId`, then check if any of the children of that clause node have the type 'conditional'.

Reply:
Where would the function be called from in TSv2?

Reply:
In the `addToContract` saga, after creating the clause node, but before inserting it into the document.

Reply:
I've added the support for detecting non-supported Slate nodes.
- This takes the editor as input but it is not used, as per previous discussion
- This should reject Slate DOMs with any of the following:
   - conditional variables
   - computed variables
   - html blocks or inlines
   - images
   - nested lists